### PR TITLE
Add more typing to event listeners

### PR DESCRIPTION
**Pull request by JackismyShephard**
## Description

This adds more typing definitions that I forgot to add in https://github.com/gradio-app/gradio/pull/8983

Closes: #8928

### CODE REVIEW

**Comment by JackismyShephard** [comment]
@freddyaboulton Should be ready for review now

File: gradio/blocks.py [hunk 1]
```diff
@@ -2162,7 +2162,10 @@ def launch(
         share: bool | None = None,
         debug: bool = False,
         max_threads: int = 40,
-        auth: Callable | tuple[str, str] | list[tuple[str, str]] | None = None,
+        auth: Callable[..., Any]
```
Comment (abidlabs) [inline gradio/blocks.py:2165]: We can be more specific here:
```suggestion
        auth: Callable[[str, str], bool]
```

**Review by abidlabs — APPROVED** [review]
lgtm with the one suggestion above

File: gradio/component_meta.py [hunk 2]
```diff
@@ -14,8 +14,10 @@
 
 INTERFACE_TEMPLATE = '''
 {{ contents }}
-    from typing import Callable, Literal
+    from typing import Callable, Literal, Sequence, Any, TYPE_CHECKING
     from gradio.blocks import Block
+    if TYPE_CHECKING:
+        from gradio.components import Timer
```
Comment (freddyaboulton) [inline gradio/component_meta.py:20]: Nice catch!

**Review by freddyaboulton — APPROVED** [review]

**Comment by JackismyShephard** [comment]
@abidlabs Should be ready for merging into main now

**Comment by abidlabs** [comment]
Nice! merging

**Pull request merged**